src/adapter/src/metrics.rs

```diff
@@ -156,10 +157,37 @@ impl Metrics {
                 var_labels: ["altered_a_cluster"],
                 buckets: histogram_seconds_buckets(0.000_128, 8.0),
             )),
+            row_set_finishing_seconds: registry.register(metric!(
+                name: "mz_row_set_finishing_seconds",
+                help: "The time it takes to run RowSetFinishing::finish.",
+                buckets: histogram_seconds_buckets(0.000_128, 16.0),
+            )),
+        }
+    }
+
+    pub(crate) fn row_set_finishing_seconds(&self) -> Histogram {
+        self.row_set_finishing_seconds.with_label_values(&[])
```

Comment:
I was going to propose we add a label here for `sorted=true|false`. Then I read the code and it looks like we always sort results even if a user didn't specify an `ORDER BY`? Although in that case it seems that no actual row movement would occur, but we'd still have to do `O(something)` to verify that. Is this correct? What's the `something` in those cases? If we are returning 1M+ rows, does the something add up even if there's no swaps?

If `something` ends up being slowish, it's worth considering delaying this PR so that can get fixed, then we can add a sorted label here.

Thoughts?

Reply:
Counterpoint against determinism at this level: having the rows returned in a deterministic order can sometimes be bad for users who don't know that, by the spec, the if there's no ORDER BY we can return in any order. They can sometimes rely on an observed order in their application code, then if we change the implementation they get bit. Go, for example, does runtime randomization of map iteration order to prevent programs from relying on something out-of-spec.

If our tests need to rely on an order, they should be using the in-spec behavior using ORDER BY and not using out-of-spec behavior.

I'm ok merging is as, but think it's bad to have things, even tests, depend on out-of-spec behavior.

Reply:
That's a great point, I filed https://github.com/MaterializeInc/materialize/issues/28261 to track and put it on the agenda for our team's weekly